tutorials/get_started/tensor_expr_get_started.py

```diff
@@ -302,18 +371,437 @@
+# .. note:: Code Specialization
+#
+#   As you may have noticed, the declarations of A, B and C all take the same
+#   shape argument, n. TVM will take advantage of this to pass only a single
+#   shape argument to the kernel, as you will find in the printed device code.
+#   This is one form of specialization.
+#
+#   On the host side, TVM will automatically generate check code that checks
+#   the constraints in the parameters. So if you pass arrays with different
+#   shapes into fadd, an error will be raised.
+#
+#   We can do more specializations. For example, we can write :code:`n =
+#   tvm.runtime.convert(1024)` instead of :code:`n = te.var("n")`, in the
+#   computation declaration. The generated function will only take vectors with
+#   length 1024.
+
+################################################################################
+# .. note:: TE Scheduling Primitives
+#
+#   TVM includes a number of different scheduling primitives:
+#
+#   - split: splits a specified axis into two axises by the defined factor.
+#   - tile: tiles will split a computation across two axes by the defined factors.
+#   - fuse: fuses two consecutive axises of one computation.
+#   - reorder: can reorder the axises of a computation into a defined order.
+#   - bind: can bind a computation to a specific thread, useful in GPU programming.
+#   - compute_at: by default, TVM will compute tensors at the root by default. comput_at specifies
+#     that one tensor should be computed at the first axis of computation for another operator.
+#   - compute_inline: when marked inline, a computation will be expanded then inserted into the
+#     address where the tensor is required.
+#   - compute_root: moves a computation to the root stage.
+#
+#   A complete description of these primitives can be found in the
+# [Schedule Primitives](https://tvm.apache.org/docs/tutorials/language/schedule_primitives.html) docs page.
+
+################################################################################
+# Example 2: Manually Optimizing Matrix Multiplication with TE
+# ------------------------------------------------------------
+#
+# Now we will consider a second, more advanced example, demonstrating how with
+# just 18 line of python code from TVM we can demonstrate up to 18x speedup on
+# a common matrix multiplication operation.
+#
+# **There are two important optimizations on intense computation applications
+# executed on CPU:**
+# 1. Increase the cache hit rate of memory access. Both complex numerical
+#    computation and hot-spot memory access can be accelerated from high cache hit
+#    rate. This requires us to transform the origin memory access pattern to the
+#    pattern fits the cache policy.
+# 2. SIMD (Single instruction multi-data), also known as the vector processing
+#    unit. Every time, a small batch of data, rather than a single grid, will be
+#    processed. This requires us to transform the data access pattern in the loop
```

Comment:
```suggestion
#    unit. On each cycle instead of processing a single value, SIMD can process a small batch of data.
#    This requires us to transform the data access pattern in the loop
```